### ISSUE TITLE
Empty geometries treated differently in GeoSeries compared to Shapely

### ISSUE THREAD

**Issue by koshy1123** [geopandas/geopandas#980]
This confused me, and I couldn't find any relevant documentation explaining this behaviour - hoping someone can shed some light on this. Notice how 2 empty geometries are "equal" in shapely land, but become "not equal" in a geoseries using the check [`.geom_equals`](http://geopandas.org/reference.html#geopandas.GeoSeries.geom_equals)
```
>>> import shapely.wkt
>>> empty1 = shapely.wkt.loads('POLYGON EMPTY')
>>> empty2 = shapely.wkt.loads('GEOMETRYCOLLECTION EMPTY')
>>> empty1.equals(empty2)
True
>>> import geopandas
>>> emptygs1 = geopandas.GeoSeries([empty1])
>>> emptygs2 = geopandas.GeoSeries([empty2])
>>> emptygs1.geom_equals(emptygs2)
0    False
dtype: bool
```
**Package versions**
```
geopandas==0.4.0
Shapely==1.6.4.post1
```

For anyone curious _why_ I was doing this, I was writing a semi-related test where I expected some empty geometries.

**Comment by koshy1123** [comment]
My theory is that it is related to:
```
>>> numpy.nan == numpy.nan
False
```

**Comment by koshy1123** [comment]
Maybe GeoSeries doesn't like `GEOMETRYCOLLECTION EMPTY`?
```
>>> geopandas.GeoSeries(empty2)
0    ()
dtype: object
>>> geopandas.GeoSeries(empty1)
0    POLYGON EMPTY
dtype: object
```

**Comment by koshy1123** [comment]
Ah, maybe related to: https://github.com/geopandas/geopandas/issues/953
GeometryCollections not being supported would explain this.

**Comment by jorisvandenbossche** [comment]
> Maybe GeoSeries doesn't like GEOMETRYCOLLECTION EMPTY?

No, the difference you see there is only a matter of the representation in a GeoSeries (which could be improved!)

> My theory is that it is related to:
>
> >>> numpy.nan == numpy.nan
> False

Yes, I think that is more or less correct. In the GeoPandas ops, it is checked whether one of both geometries are empty, and then it is considered False (and missing values are handled by considering them as empty). So at this moment, in the operations, we don't really distinguish empty or missing (although I hope to change that in the future, see https://github.com/geopandas/geopandas/issues/601).

But, the (possibly unintended) consequence is indeed this inconsistency between the result for individual shapely operations and geopandas GeoSeries.

**Comment by koshy1123** [comment]
Looks like this one is fixed with GeoPandas 0.6+. I'm closing this one out @jorisvandenbossche
